pkg/scheduler/queue/queue.go

```diff
@@ -377,31 +368,35 @@ func (q *RequestQueue) GetNextRequestForQuerier(ctx context.Context, last UserIn
 }
 
 func (q *RequestQueue) stop(_ error) error {
-	q.stopRequested <- struct{}{} // Why not close the channel? We only want to trigger dispatcherLoop() once.
+	close(q.stopRequested)
```

Comment:
I don't think this is equivalent: closing the channel means that the loop in `dispatcherLoop` will run continuously, as the `select` call will always be able to choose the `stopRequested` branch. https://go.dev/play/p/kmJbw_C-s5d demonstrates this behaviour.

Reply:
:+1:  good call, I see we want to keep looping, dispatching queries, and even processing any querier operations.

Will add a note on why this works this way